projects/js-packages/licensing/components/activation-screen-error/use-get-error-content.tsx

```diff
@@ -0,0 +1,185 @@
+import { getRedirectUrl } from '@automattic/jetpack-components';
+import { ExternalLink } from '@wordpress/components';
+import { createInterpolateElement } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import React from 'react';
+import { LICENSE_ERRORS } from './constants';
+
+type LicenseErrorKeysType = keyof typeof LICENSE_ERRORS;
+type LicenseErrorValuesType = ( typeof LICENSE_ERRORS )[ LicenseErrorKeysType ];
+
+export const UseGetErrorContent = ( licenseError: string, errorType: LicenseErrorValuesType ) => {
+	const hasLicenseError = licenseError !== null && licenseError !== undefined;
+
+	if ( ! hasLicenseError ) {
+		return {
+			errorMessage: null,
+			errorInfo: null,
+		};
+	}
+
+	const needHelpGetInTouchLink = createInterpolateElement(
+		__( 'Need help? <a>Get in touch</a>.', 'jetpack' ),
+		{
+			a: (
+				<ExternalLink
+					href={ getRedirectUrl( 'jetpack-support-license-activation' ) }
+					rel="noopener noreferrer"
+				></ExternalLink>
+			),
+		}
+	);
+
+	switch ( errorType ) {
+		case LICENSE_ERRORS.NOT_SAME_OWNER:
+			return {
+				errorMessage: __(
+					'The account that purchased the plan and the account managing this site are different.',
```

Comment:
I think this sounds needlessly complicated personally. On the support page we link to here, the error is `This license key is owned by a different user.` Perhaps we could go with that instead?

Reply:
Yeah this ⬆️ is the error message that displayed prior to this PR, and it is specifically the error that is resulting in regular support tickets and churn. 
Because ideally what we need to get across to the user is:  the (wordpress.com) user account that their site is using for the Jetpack connection is not the same (wordpress.com) user account that was used to purchase the product/license.
You see, there has been various discussion about this: p1HpG7-rfm-p2, and there is an upcoming project: pbNhbs-aGG-p2 to more throughly handle this issue, but for now as quick improvement, this is what design came up with.
Here are my thoughts:  TBH, I agree that this error copy could be improved, but at the same time, I'm not necessarily eager to strike up a back and forth feedback loop of copy suggestion, because 1. This is merely a maintenance task, not a project, 2. there has already been quite a bit of discussion about this, 3. this will all be changed or removed in the near future when we are able to work on the upcoming planned project, and 4. Regardless of whether we could improve on the error message or not, I believe all these changes as a whole are quite a good improvement (with limited time investment), especially with the additional help steps/hints that go along with the error.